### DISCUSSION TITLE
Query regarding the self collision API

### DISCUSSION THREAD

**Discussion by shannonwarren** [NVlabs/curobo#232] [Software Q&A] [answered]
I had a really basic question. I'm looking at the example for collision checking and I see we pass ` q_s = curobo_fn.sample(6, mask_valid=False)` to `d_world, d_self = curobo_fn.get_world_self_collision_distance_from_joints(cu_js.position)`. is 6 the number of randomly generated joint states it passes to do the collision checking given the robot config and world? I'm a bit confused here. What's the sample call doing and what if I just want to check the world collision of the start and end state?

```
This is my output of q_s
tensor([[-3.7920, -4.8940, -0.8224,  0.9245, -5.8487, -0.1717],
        [ 1.6183,  2.8923,  2.3408,  2.4027, -1.4540,  5.4627],
        [-5.0721,  1.5522, -0.1606, -3.0105, -4.0277,  4.5790],
        [-1.7511,  3.9157, -1.1222,  4.6432,  1.8629,  5.0405],
        [ 1.9021, -1.2694, -2.2688, -4.3548,  0.0798,  3.7516],
        [-4.7400,  3.6330, -1.6245, -1.9392,  2.0402, -5.4627]],
```

**Comment by balakumar-s** [comment, marked as answer]
The sample call is just sampling random joint configurations. You can replace that with your start and end state to do collision checking. 

> is 6 the number of randomly generated joint states it passes to do the collision checking given the robot config and world?

Yes.
